src/components/todo-list/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from "@/utils/core/hooks";
+import { IVDOMNode } from "@/types/vdom";
+import Todo from "@/components/todo-list/todo";
+import { ITodo } from "@/types/todo-list";
+
+export default function TodoList(): IVDOMNode {
+  const [todos, setTodos] = useState<ITodo[]>([]);
+  const [input, setInput] = useState("");
+
+  const inputText = (e: { target: HTMLInputElement }) => {
+    setInput(e.target.value);
+  };
+
+  const addTodo = () => {
+    if (!input.trim()) return;
+    setTodos([...todos, { id: Date.now(), text: input, completed: false }]);
+    setInput("");
+  };
+
+  const toggleTodo = (id: number) => {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      )
+    );
```

Comment:
질문

setTodos를 아래처럼도 쓸 수 있는데요.
```suggestion
    setTodos(
      todos.map((prevTodos) =>
        prevTodos.id === id ? { ...todo, completed: !todo.completed } : prevTodos
      )
    );
```

외부에서 state를 가져와 사용하는 것과 prev state를 가져와 사용하는 것 이 둘의 차이에 대해 알고 계신가요!?

Reply:
아하 넵 어떤걸 우려하시는지는 알것 같습니다.
그리고 차이점은 적당한... 수준 정도로만 이해하고있는데 명확히 설명이 어렵네요
좀 이상한 설명인가 싶기도 하지만 쉽게말하면 리렌더링보다 광클이 더 빠를 경우 생기는 문제를 해결하고싶을때 쓰면 좋습니다.
수정해주신 코드도 이론적으로 토글이나 딜리트를 리렌더보다 빠르게 한다면 의도치못한 스테이트 변경이 발생할 수는 있지만 크게 신경쓸 수준이라고 판단하진 않았습니다.
`
const [count, setCount] = useState(0);

setCount(count + 1);       
setCount(count + 1);       
// count: 1

setCount(prev => prev + 1); 
setCount(prev => prev + 1); 
// count:2
`
그리고 스테이트 내부 로직 공부한걸로 좀더 생각해보았는데요, 두 코드의 결과가 다른 부분으로 설명해보면(광클대신 연달아 실행)
아마 전자는 셋스테이트 내부 함수에서 해당 렌더링 시점의 카운트를 원시값으로 가져와서 실행하니 0+1이 두번 할당되어서 1이 되는것 같구요
후자는 내부함수에서 함수를 즉시 실행해서 값을 리턴하니 실행될때마다 업데이트된 prev로 count가 의도대로 0+1, 1+1이 되어서 2가 되는것 같네요